### PR TITLE
Demultiplex ABCs in one step using cutadapt.

### PR DESCRIPTION
**Pull request by pontushojer**
Changed so that ABC can be demultiplexed in one step using cutadapt. Before the trimmed-abc.fastq needed to be read for each ABC sequence. This introduced possible errors as ABC could be linked to more than one ABC. 

The solution required a fasta file with ABC sequences as input. These also need to be anchored (prefixed with '^') to work properly with cutadapt. 

Also changed so that all ABC related files end up in separate folder called `ABCs`.

### CODE REVIEW

File: construct-info/ABC-sequences.fasta [hunk 1]
```diff
@@ -0,0 +1,6 @@
+>ABC1
+^GCGTA
+>ABC2
+^ATAGC
+>ABC3
+^GTGCA
```
Comment (FrickTobias) [inline construct-info/ABC-sequences.fasta]: Writing `>` in the sequence is not something included in fasta format, I would suggest adding this in the get_abcs function.

File: src/dbspro/utils.py [hunk 2]
```diff
@@ -6,10 +6,30 @@
...
 
 
+def get_abcs(abc_fasta_file):
+    """
+    Helper function to get ABC sequences and names into pandas dataframe
+    :param abc_fasta_file:
+    :return: dataframe:
+    """
+    with dnaio.open(abc_fasta_file, fileformat="fasta", mode="r") as abc_fasta:
+        abc = pd.DataFrame([{"Sequence": entry.sequence, "Target": entry.name} for entry in abc_fasta])
```
Comment (FrickTobias) [inline src/dbspro/utils.py]: If we instead add a `^` to normal fasta sequences here we a) follow fasta format, b) ensure it's impossible to not give anchored sequences and c) can remove rows 24-28.

File: construct-info/ABC-sequences.fasta (hunk 1 above)
Comment (pontushojer) [reply construct-info/ABC-sequences.fasta]: Now I don't quite follow, do you mean that we should add '>' in the get_abc function and keep it out of the original file?

File: src/dbspro/utils.py (hunk 2 above)
Comment (pontushojer) [reply src/dbspro/utils.py]: So the issue here is that the cutadapt requires a fasta file with the adaptors to demultiplex. Thus I cannot supply the sequence modified with the anchor '^' directly. What I could do however is to read in the fasta and make a copy of if that guarantees the anchor modification.

**Pull request merged**